### ISSUE TITLE
Unable to parse config file

### ISSUE THREAD

**Issue by tiredofit** [nextcloud/notify_push#19]
- Notify_Push 0.12 from tarball, installed in Alpine Linux 3.13 - x86_64

````bash
bash-5.1# ./notify_push /www/nextcloud/config/config.php
Error: 
   0: Failed to load config from nextcloud config file
   1: Failed to parse config file
   2: No valid token found, expected one of [Null, Bool, LiteralString, Integer, Float, Array, SquareOpen]

Location:
   src/config.rs:41

Backtrace omitted.
Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.
````

````bash
<?php
$CONFIG = array (
  'instanceid' => '12345',
  'datadirectory' => '/data/userdata/',
  'apps_paths' => 
  array (
    0 => 
    array (
      'path' => '/www/nextcloud/apps',
      'url' => '/apps',
      'writable' => false,
    ),
    1 => 
    array (
      'path' => '/data/apps/',
      'url' => '/custom-apps',
      'writable' => true,
    ),
  ),
  'passwordsalt' => '12345',
  'secret' => '12345',
  'trusted_domains' => 
  array (
    0 => 'localhost',
    1 => 'nextcloud.example.com',
  ),
  'dbtype' => 'mysql',
  'version' => '21.0.0.18',
  'overwrite.cli.url' => 'http://localhost',
  'dbname' => 'nextcloud',
  'dbhost' => 'nextcloud-example-com-db',
  'dbport' => '3306',
  'dbtableprefix' => 'oc_',
  'mysql.utf8mb4' => true,
  'dbuser' => 'nextcloud',
  'dbpassword' => 'mysqlpassword',
  'installed' => true,
  'maintenance' => false,
  'allow_user_to_change_display_name' => false,
  'appcodechecker' => true,
  'appstoreenabled' => true,
  'auth.bruteforce.protection.enabled' => true,
  'cache_chunk_gc_ttl' => '60*60*24',
  'cache_path' => '/data/cache/',
  'check_data_directory_permissions' => true,
  'check_for_working_htaccess' => false,
  'check_for_working_wellknown_setup' => true,
  'cipher' => 'AES-256-CFB',
  'debug' => false,
  'default_language' => 'en',
  'default_locale' => 'en_US',
  'defaultapp' => '',
  'enable_previews' => false,
  'filelocking.debug' => false,
  'filelocking.enabled' => true,
  'filesystem_check_changes' => 0,
  'force_language' => 'en',
  'force_locale' => 'en_US',
  'forwarded_for_headers' => 
  array (
    0 => 'HTTP_X_FORWARDED',
    1 => 'HTTP_FORWARDED_FOR',
  ),
  'hashingCost' => 10,
  'knowledgebaseenabled' => false,
  'ldapUserCleanupInterval' => 51,
  'log_query' => true,
  'log_type' => 'file',
  'logdateformat' => 'Y-m-d H:i:s',
  'logfile' => '/www/logs/nextcloud//nextcloud.log',
  'logfilemode' => 416,
  'login_form_autocomplete' => true,
  'loglevel' => 2,
  'logtimezone' => 'America/Vancouver',
  'mail_domain' => 'wxample.com',
  'mail_from_address' => 'noreply',
  'mail_send_plaintext_only' => false,
  'mail_sendmailmode' => 'smtp',
  'mail_smtpauth' => false,
  'mail_smtpauthtype' => 'none',
  'mail_smtpdebug' => false,
  'mail_smtphost' => 'postfix-relay',
  'mail_smtpmode' => 'smtp',
  'mail_smtpname' => '',
  'mail_smtppassword' => '',
  'mail_smtpport' => 25,
  'mail_smtpsecure' => '',
  'mail_smtptimeout' => 10,
  'max_filesize_animated_gifs_public_sharing' => 10,
  'overwritehost' => '',
  'overwriteprotocol' => 'https',
  'part_file_in_storage' => true,
  'preview_libreoffice_path' => '/usr/bin/libreoffice',
  'preview_max_filesize_image' => 50,
  'preview_max_x' => 100,
  'preview_max_y' => 100,
  'session_keepalive' => true,
  'sharing.maxAutocompleteResults' => 0,
  'sharing.minSearchStringLength' => 0,
  'sharing.enable_share_accept' => false,
  'sharing.force_share_accept' => false,
  'sharing.enable_share_mail' => false,
  'simpleSignUpLink.shown' => false,
  'skeletondirectory' => '/data/templates/skeleton',
  'sort_groups_by_name' => false,
  'sqlite.journal_mode' => 'DELETE',
  'tempdirectory' => '/data/tmp/',
  'theme' => '',
  'token_auth_enforced' => false,
  'trashbin_retention_obligation' => 'auto',
  'updatechecker' => true,
  'upgrade.disable-web' => false,
  'versions_retention_obligation' => 'auto',
  'trusted_proxies' => 
  array (
    1 => '172.16.0.0/12',
  ),
  'memcache.local' => '\\OC\\Memcache\\Redis',
  'memcache.locking' => '\\OC\\Memcache\\Redis',
  'memcache.distributed' => '\\OC\\Memcache\\Redis',
  'redis' => 
  array (
    'host' => 'redis',
    'port' => 6379,
    'timeout' => 0.0,
    'password' => '',
    'dbindex' => 0,
  ),
  'ldapIgnoreNamingRules' => false,
  'ldapProviderFactory' => 'OCA\\User_LDAP\\LDAPProviderFactory',
  'default_phone_region' => 'us',
);
````

**Comment by icewind1991** [comment]
Strange, it has no issue parsing this config for me.

0.1.3 should give a better error message when the parsing fails.

**Comment by tiredofit** [comment]
Thanks for the updated build. It seems that a line break between <?php and the `CONFIG` array declaration are throwing errors for me.

````
bash-5.1# /opt/nextcloud_files_backend/bin/x86_64/notify_push /www/nextcloud/config/config.php
Error: 
   0: Failed to load config from nextcloud config file
   1: Failed to parse config file
   2:  
      1 | <?php                                                                                                  
        | ^^ No valid token found, expected one of [Null, Bool, LiteralString, Integer, Float, Array, SquareOpen]
      2 |
      3 | $CONFIG = array(
````

With a comment: 
````
bash-5.1# /opt/nextcloud_files_backend/bin/x86_64/notify_push /www/nextcloud/config/config.php
Error: 
   0: Failed to load config from nextcloud config file
   1: Failed to parse config file
   2:  
      1 | <?php                                                                                                  
        | ^^ No valid token found, expected one of [Null, Bool, LiteralString, Integer, Float, Array, SquareOpen]
      2 | /** Test **/
      3 | $CONFIG = array(
````

Removing the blank line it will then parse fine.

**Comment by icewind1991** [comment]
Should be fixed with 0.1.4
